### PR TITLE
Raise an error when creating a task but M2 was called w/ --no-threads

### PR DESCRIPTION
**Pull request by d-torrance**
### Before

We could create tasks, but they'd never be run:

```m2
$ M2 --silent --no-threads

i1 : t = schedule(() -> 3);

i2 : sleep 5

o2 = 0

i3 : isReady t

o3 = false
```

### After

```m2
$ M2 --silent --no-threads

i1 : t = schedule(() -> 3);
stdio:1:12:(3): error: thread support has been disabled
```

Closes: #3570 

---
I also noticed another bug while working on this.  Previously, if we call `taskResult` on a `Task` object that was created with `taskResult` but we never called `schedule`, then Macaulay2 would hang:

```m2
i1 : t = createTask(() -> 3);

i2 : taskResult t
^C^C
Exit (y=yes/n=no/a=abort/b=backtrace)? b
-* stack trace, pid: 3305389
 0# stack_trace(std::ostream&, bool) at /usr/src/macaulay2-1.24.05+git202410290107-0ppa202410281907~ubuntu24.04.1/M2/Macaulay2/d/main.cpp:136
 1# interrupt_handler at /usr/src/macaulay2-1.24.05+git202410290107-0ppa202410281907~ubuntu24.04.1/M2/Macaulay2/d/main.cpp:300
 2# 0x00007CA6F4A45320 in /lib/x86_64-linux-gnu/libc.so.6
 3# __GI___futex_abstimed_wait_cancelable64 at ./nptl/futex-internal.c:139
 4# pthread_cond_wait at ./nptl/pthread_cond_wait.c:627
 5# ThreadTask::waitOn() at /usr/src/macaulay2-1.24.05+git202410290107-0ppa202410281907~ubuntu24.04.1/M2/Macaulay2/system/supervisor.cpp:211
 6# taskResult_1 at /usr/src/macaulay2-1.24.05+git202410290107-0ppa202410281907~ubuntu24.04.1/M2/Macaulay2/d/pthread-tmp.c:465
```

Now we raise an error instead:

```m2
i1 : t = createTask(() -> 3);

i2 : taskResult t
stdio:2:10:(3): error: task not scheduled yet
```

### CODE REVIEW

File: M2/Macaulay2/d/pthread.d [hunk 1]
```diff
@@ -97,8 +98,12 @@ nextTaskSerialNumber():int := (
      taskSerialNumber = taskSerialNumber+1;		    -- race condition here, solve later
      taskSerialNumber);
 
+NoThreadsError() ::= buildErrorPacket("thread support has been disabled");
+
 createTask2(fun:Expr,arg:Expr):Expr :=(
```
Comment (mahrud) [inline M2/Macaulay2/d/pthread.d:103]: Why is this called `createTask2`?

**Comment by mahrud** [comment]
It might be worth adding a couple of tests.

**Review by mahrud — APPROVED** [review]

File: M2/Macaulay2/d/pthread.d (hunk 1 above)
Comment (d-torrance) [reply M2/Macaulay2/d/pthread.d:103]: I think because it takes two arguments?

**Comment by d-torrance** [comment]
> It might be worth adding a couple of tests.

Definitely!  The thread documentation could certainly use some work -- see also #1147.

**Pull request merged**